runtime/grid_reduction.cu

```diff
@@ -683,7 +713,7 @@ __device__ void iterGroupedGridReduceLastBlock(
         grid_reduction_segment_size, // Number of reductions across
                                      // grid reduce dimensions
     const nvfuser_index_t
-        block_reduction_segment_size, // Number of reductions across the block
```

Comment:
In the code, `grid_segment_size` represents `number of reductions in the grid, e.g. gdimx`
`grid_reduction_segment_size` represents `Number of reductions across grid reduce dim, e.g. gdimy`
similarly, we should use `block_segment_size` to represent `Number of reductions across the block, e.g. bdimx`. 
so here, this `block_reduction_segment_size` is renamed to `block_segment_size`, just a rename, no change in logic.